src/main/java/in/zapr/druid/druidry/client/DruidJerseyClient.java

```diff
@@ -0,0 +1,131 @@
+package in.zapr.druid.druidry.client;
+
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.glassfish.jersey.apache.connector.ApacheClientProperties;
+import org.glassfish.jersey.apache.connector.ApacheConnectorProvider;
+import org.glassfish.jersey.client.ClientConfig;
+
+import java.util.List;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import in.zapr.druid.druidry.client.exception.ConnectionException;
+import in.zapr.druid.druidry.client.exception.QueryException;
+import in.zapr.druid.druidry.query.DruidQuery;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DruidJerseyClient implements DruidClient {
+
+    private static final int DEFAULT_CONNECTION_POOL_LIMIT = 5;
```

Comment:
Just to know : Ideally the connection pool size could be decided by number of cores and the RAM size.

Reply:
Will see it in future version